index.d.ts

```diff
@@ -10,226 +10,233 @@ declare global {
+   */
+  encrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
+  ): Promise<Uint8Array>;
+
+  /**
+   * Decrypt a string encrypted with the user's wallet.
+   *
+   * @param data Buffer-like structure to decrypt.
+   * @param options Decrypt options. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/decrypt#algorithm.
+   *
+   * @returns Promise of the decrypted `Uint8Array`.
+   */
+  decrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
+  ): Promise<Uint8Array>;
+
+  /**
+   * Get the user's custom Arweave config set in the extension
+   *
+   * @returns Promise of the user's Arweave config
+   */
+  getArweaveConfig(): Promise<GatewayConfig>;
+
+  /**
+   * @deprecated Find alternatives at https://docs.arconnect.io/api/signature
+   */
+  signature(
+    data: BinaryDataType,
+    // https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign#algorithm
+    algorithm?: RsaPssParams | EcdsaParams | { name: "RSASSA-PKCS1-v1_5" } | { name: "HMAC" },
+  ): Promise<Uint8Array>;
+
+  /**
+   * Get the user's active public key, from their wallet
+   *
+   * @returns Promise of the active public key
+   */
+  getActivePublicKey(): Promise<string>;
+
+  /**
+   * Add a token to ArConnect (if it is not already present)
+   *
+   * @param id Token contract ID
+   * @param type Type of the token (asset or collectible)
+   * @param gateway Gateway config for the token
+   */
+  addToken(id: string, type?: TokenType, gateway?: GatewayConfig): Promise<void>;
+
+  /**
+   * Checks if a token has been added to ArConnect
+   * 
+   * @param id Token to check for
+   * 
+   * @returns Token added or not
+   */
+  isTokenAdded(id: string): Promise<boolean>;
+
+  /**
+   * Dispatch an Arweave transaction (preferably bundled)
+   *
+   * @param transaction Transaction to dispatch
+   * 
+   * @returns Dispatched transaction ID and type
+   */
+  dispatch(transaction: Transaction): Promise<DispatchResult>;
+
+  /**
+   * Create a deterministic secret based on the input data.
+   * 
+   * @param data Input data to generate the hash from
+   * @param options Hash configuration
+   * 
+   * @returns Hash array
+   */
+  privateHash(data: ArrayBuffer, options: SignMessageOptions): Promise<Uint8Array>; // TODO: data is ArrayBuffer or BinaryDataType
+
+  /**
+   * Create and sign a DataItem (bundled transaction item),
+   * that can be loaded into "arbundles".
+   * 
+   * @param dataItem Data item params
+   * 
+   * @returns Buffer of the signed data item
+   */
+  signDataItem(dataItem: DataItem): Promise<ArrayBufferLike>; // TODO: Is this ArrayBufferLike or Buffer based on the example? `const dataItem = new DataItem(Buffer.from(result));` The implementation says `return Array.from<number>(dataEntry.getRaw());` in https://github.com/arconnectio/ArConnect/blob/production/src/api/modules/sign_data_item/sign_data_item.background.ts
```

Comment:
I didn't change these types in the PR, but see my notes:

- The example in the docs take the returned value and pass it directly to `DataItem()`, but `DataItem` only takes `Buffer`, so that's wrong (type-wise):

    ![image](https://github.com/user-attachments/assets/71bbedaf-edbb-4281-a0c6-bb6f2e355c56)

- However, the implementation says `return Array.from<number>(dataEntry.getRaw())`, which is not the same as `ArrayBufferLike`, but happens to pass as `Buffer`.

In this case, in Othent, I just made sure I'm returning `ArrayBuffer` (`return dataItemInstance.getRaw().buffer`) and transformed it before passing it to `DataItem`: `new DataItem(Buffer.from(result))`. Alternatively, this could just return `Buffer` (`return dataItemInstance.getRaw()`) and the example would be valid.